src/Collapse/include/gudhi/Flag_complex_edge_collapser.h

```diff
@@ -0,0 +1,375 @@
+    IVertex rw_u = insert_vertex(u);
+    IVertex rw_v = insert_vertex(v);
+#ifdef DEBUG_TRACES
+    std::cout << "Inserting the edge " << u <<", " << v << std::endl;
+#endif  // DEBUG_TRACES
+    sparse_row_adjacency_matrix_[rw_u].insert(rw_v) = idx;
+    sparse_row_adjacency_matrix_[rw_v].insert(rw_u) = idx;
+    return std::minmax(rw_u, rw_v);
+  }
+
+ public:
+  /** \brief Flag_complex_edge_collapser constructor from a range of filtered edges.
+   *
+   * @param[in] edges Range of Filtered edges range.There is no need the range to be sorted, as it will be performed in
+   * `Flag_complex_edge_collapser::process_edges`.
+   *
+   * \tparam FilteredEdgeRange must be a range for which std::begin and std::end return iterators on a
+   * `Flag_complex_edge_collapser::Filtered_edge`.
+   */
+  template<typename FilteredEdgeRange>
+  Flag_complex_edge_collapser(const FilteredEdgeRange& edges)
+  : f_edge_vector_(std::begin(edges), std::end(edges)) { }
+
+  /** \brief Performs edge collapse in a increasing sequence of the filtration value.
+   *
+   * \tparam filtered_edge_output is a functor that is called on the output edges, in non-decreasing order of
+   * filtration, as filtered_edge_output(u, v, f) where u and v are Vertex_handle representing the extremities of the
+   * edge, and f is its new Filtration_value.
+   */
+  template<typename FilteredEdgeOutput>
+  void process_edges(FilteredEdgeOutput filtered_edge_output) {
+    // Sort edges
+    auto sort_by_filtration = [](const Filtered_edge& edge_a, const Filtered_edge& edge_b) -> bool
+    {
+      return (std::get<2>(edge_a) < std::get<2>(edge_b)); 
+    };
+
+#ifdef GUDHI_USE_TBB
+    tbb::parallel_sort(f_edge_vector_.begin(), f_edge_vector_.end(), sort_by_filtration);
+#else
+    std::sort(f_edge_vector_.begin(), f_edge_vector_.end(), sort_by_filtration);
+#endif
+
+    for (Edge_index endIdx = 0; endIdx < f_edge_vector_.size(); endIdx++) {
+      Filtered_edge fec = f_edge_vector_[endIdx];
+      Vertex_handle u = std::get<0>(fec);
+      Vertex_handle v = std::get<1>(fec);
+      Filtration_value filt = std::get<2>(fec);
+
+      // Inserts the edge in the sparse matrix to update the graph (G_i)
+      IEdge ie = insert_new_edge(u, v, endIdx);
+
+      iedge_to_index_map_.emplace(ie, endIdx);
+      critical_edge_indicator_.push_back(false);
+
+      if (!edge_is_dominated(u, v)) {
+        critical_edge_indicator_[endIdx] = true;
+        filtered_edge_output(u, v, filt);
+        if (endIdx > 1)
+          set_edge_critical(endIdx, filt, filtered_edge_output);
+      }
+    }
+  }
+
+};
+
+/** \brief Constructs a Flag complex from edges as an input, collapse edges and returns remaining edges.
+ *
+ * \fn auto Gudhi::collapse::flag_complex_collapse_edges(FilteredEdgeRange const& edges)
+ *
+ * \tparam FilteredEdgeRange furnishes `std::begin` and `std::end` methods and returns an iterator on a
+ * FilteredEdge of type `std::tuple<Vertex_handle, Vertex_handle, Filtration_value>`
```

Comment:
I think we said the doc should speak about the edges being unique. We may want to mention if the range needs to be already sorted. We have lost some information when moving from the class to the function.

Reply:
I did it on d8c5a1b2